Main.cpp

```diff
@@ -163,29 +160,22 @@ int InitProc()
 	return 1; // return 1 if OK; 0 on failure
 }
 
-void AIProc()
+Export void AIProc()
 {
 }
 
-void __cdecl GetSaveRegions(struct BufferDesc &bufDesc)
-{
-    // Note: this has been changed. The count must be persisted in the saved game file,
-    // so that the AI is in the right state when the game is reloaded.
-
-	bufDesc.bufferStart = &saveData;     // Pointer to a buffer that needs to be saved
-	bufDesc.length = sizeof(saveData);	// sizeof(buffer)
-}
+ExportSaveLoadData(saveData);
 
-int StatusProc()
+Export int StatusProc()
 {
 	return 0; // must return 0
 }
 
-SCRIPT_API void NoResponseToTrigger()
+Export void NoResponseToTrigger()
 {
 }
 
-SCRIPT_API void __cdecl AIStateChange()
+Export void __cdecl AIStateChange()

```

Comment:
I don't know if `__cdecl` is really needed on functions such as this one. There are no arguments to this function, so it shouldn't matter how arguments are passed, or who cleans up the arguments on the stack. It's not wrong to have it though. Other similar free standing functions with arguments are `__cdecl`, such as `GetSaveRegions`. It's just that's it's probably more detail than people need to worry about, and I don't think other projects bother with it.

Actually, the only methods that need `__cdecl` (freestanding, with arguments) are the exported `GetSaveRegions` method, where the detail is hidden away in a macro, and a couple of imported functions, such as `CreateSetTrigger` and `scr_snprintf`, where the detail is hidden away in the import header. None of the other DLL export methods take arguments, be they standard required methods, or trigger callbacks, so it's just not relevant. It's not even relevant as a model for other similar methods, unless someone hand codes `GetSaveRegions`.